src/client.rs

```diff
@@ -522,3 +526,17 @@ impl Client {
         }
     }
 }
+
+#[cfg(test)]
+mod client_tests {
+    use super::Client;
+    use crate::Url;
+
+    #[test]
+    fn base_url() {
+        let mut client = Client::new();
+        client.set_base_url(Url::parse("http://example.com/api/v1/").unwrap());
+        let url = client.url("posts.json");
+        assert_eq!(url.as_str(), "http://example.com/api/v1/posts.json");
+    }
```

Comment:
I debated putting this assertion in the documentation test, but they were all already marked as `no_run`. Just let me know where you prefer!